rails/spec/requests/admin_user_request_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+RSpec.describe "SuperAdmin user request", type: :request do
+  describe "GET edit" do
+    context "with a super admin privileges" do
+      let(:super_admin) { FactoryBot.create(:user, role: 100, super_admin: true) }
+      let(:user) { FactoryBot.create(:user, email: "user@mail.co", username: "user_name", role: 2) }
+
+      before do
+        login_as super_admin 
+      end
+
+      it "should return success if the user is an admin" do
```

Comment:
Let's also add a spec for when the logged-in user is a super admin, but the user itself is not an admin. The controller accounts for this with a redirect, but that is not encapsulated in a spec here.